.travis.yml

```diff
@@ -193,12 +209,19 @@ install:
 
 # TODO (2016-02-21): Split the long lines (`\` doesn't work).
 
+# We have to manually install `alex`, `cpphs` and possibly `happy`
+# for older snapshots because `stack` refuses to install build tools
+# outside the designated snapshot. [Issue commercialhaskell/stack #595]
+
 # N.B. that we use the `--force-reinstalls` option [Issue 1520].
 
   - if [[ $TEST = "MAIN" || $TEST = "HADDOCK" ]]; then
        make CABAL_OPTS='--only-dependencies --force-reinstalls' install-bin;
     elif [[ $TEST = "STACKAGE" ]]; then
-         stack build $ARGS --no-terminal --test --only-dependencies;
+       if [[ $MANUAL_INSTALL_BUILD_TOOLS = "YES" ]]; then
+           stack install $ARGS --no-terminal alex cpphs happy;
```

Comment:
The issue https://github.com/commercialhaskell/stack/issues/1427 seems related, and then it points back to https://github.com/commercialhaskell/stack/issues/595. Thus I leave the comments like that. I do not know what is wrong with the resolver for build tools. In particular, I did not specify the versions of happy and Alex in the `yaml` file.